Add ignore.tags into a configuration.

fujiwara opened the pull request:
refs #726 

This PR adds `ignore.tags` into a configuration.

For example, set "ecspresso:ignore" to `ignore.tags` as below,

```yaml
ignore:
  tags:
    - ecspresso:ignore # ignore tags of service and task definition
```

ecspresso ignores the tag that the key equals "ecspresso:ignore" of ECS services and task definitions.

Even if the tag exists in definition files or resources in ECS, ecspresso does not edit the tags.

This is useful for managing some tags with other tools.

fujiwara committed:
add ignore section into config.

refs #726

```yaml
ignore:
  tags:
    - foo
```

fujiwara committed:
add tests/ci files

fujiwara committed:
update readme for ignore.

ponkio-o commented on config_test.go:
```diff
@@ -323,3 +337,57 @@ func TestFilterCommandDeprecated(t *testing.T) {
+	resourceTags []types.Tag
+	expectedTags []types.Tag
+}{
+	{
```
[nits] I thought it would be better to add test case to check case-sensitive.

> Tag keys and values are case-sensitive.

https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateService.html#ECS-CreateService-request-tags

fujiwara committed:
fix tests

add test case for case sensitive.

fujiwara committed:
refactoring

Add hasTag() interface.
TaskDefinitionInput is a new type from ecs.RegisterTaskDefinitionInput.

fujiwara committed:
refactoring

TaskDefinition is a new type.

fujiwara committed:
not need to type alias
